src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -3838,9 +3839,39 @@ public Logic(SourceShape<T> shape, IAsyncEnumerable<T> enumerable) : base(shape)
             public override void PreStart()
             {
                 base.PreStart();
-                _completionCts = new CancellationTokenSource();
                 _enumerator = _enumerable.GetAsyncEnumerator(_completionCts.Token);
             }
+            
+            public override void PostStop()
+            {
+                try
+                {
+                    _completionCts.Cancel();
+                    _completionCts.Dispose();
+                }
+                catch(Exception ex)
+                {
+                    // This should never happen
+                    Log.Error(ex, "AsyncEnumerable threw while cancelling CancellationTokenSource");
+                }
+                try
+                {
+                    _enumerator.DisposeAsync().AsTask().ContinueWith(t =>
```

Comment:
Nitpick: why not just have a detatched `Task` running in a local function which `await`s on `DisposeAsync` ? You can place the entire `try...catch` inside of that and have cleaner error-handling semantics.

Reply:
done